sdk/eventhub/azure-eventhub-checkpointstoretable/azure/eventhub/extensions/checkpointstoretable/_tablestoragecs.py

```diff
@@ -22,17 +27,217 @@ class TableCheckpointStore:
+                  The format is like "<namespace>.servicebus.windows.net".
+                - `eventhub_name` (str): The name of the specific Event Hub the checkpoints are associated with,
+                  relative to the Event Hubs namespace that contains it.
+                - `consumer_group` (str): The name of the consumer group the checkpoints are associated with.
+                - `partition_id` (str): The partition ID which the checkpoint is created for.
+                - `sequence_number` (int): The sequence number of the :class:`EventData<azure.eventhub.EventData>`.
+                - `offset` (str): The offset of the :class:`EventData<azure.eventhub.EventData>`.
+        """
+        thePartitionKey = "{} {} {} Checkpoint".format(fully_qualified_namespace,
+        eventhub_name, consumer_group)
+        my_filter = "PartitionKey eq '"+ thePartitionKey + "'"
+        entities = self.table_client.query_entities(my_filter)
+        checkpointslist = []
+        for entity in entities:
+            dic = {}
+            dic[u'sequencenumber'] = entity[u'sequencenumber']
+            dic[u'offset'] = entity[u'offset']
+            checkpointslist.append(dic)
+        return checkpointslist
+
+    def update_checkpoint(self, checkpoint):
+        """Updates the checkpoint using the given information for the offset, associated partition and
+        consumer group in the storage table.
+        Note: If you plan to implement a custom checkpoint store with the intention of running between
+        cross-language EventHubs SDKs, it is recommended to persist the offset value as an string.
+        :param Dict[str,Any] checkpoint: A dict containing checkpoint information:
+                - `fully_qualified_namespace` (str): The fully qualified namespace that the Event Hub belongs to.
+                  The format is like "<namespace>.servicebus.windows.net".
+                - `eventhub_name` (str): The name of the specific Event Hub the checkpoint is associated with,
+                  relative to the Event Hubs namespace that contains it.
+                - `consumer_group` (str): The name of the consumer group the checkpoint is associated with.
+                - `partition_id` (str): The partition ID which the checkpoint is created for.
+                - `sequence_number` (int): The sequence number of the :class:`EventData<azure.eventhub.EventData>`
+                  the new checkpoint will be associated with.
+                - `offset` (str): The offset of the :class:`EventData<azure.eventhub.EventData>`
+                  the new checkpoint will be associated with.
+        :rtype: None
+        """
+        try:
+            checkpoint_entity = self._create_checkpoint_entity(checkpoint)
+            self.table_client.update_entity(mode=UpdateMode.REPLACE, entity=checkpoint_entity)
+        except ResourceNotFoundError:
+            self._checkpoint_entity(checkpoint)
```

Comment:
Do we need `_checkpoint_entity`? It seems like you could change the code to something like this:

```python
checkpoint_entity = self._create_checkpoint_entity(checkpoint)
try:
    self.table_client.update_entity(mode=UpdateMode.REPLACE, entity=checkpoint_entity)
except ResourceNotFoundError:
    self.table_client.create_entity(entity=checkpoint_entity)
```

Reply:
Yea that is a great catch that change looks like it will make the code cleaner